src/program.tsx

```diff
@@ -69,10 +69,11 @@ class Program extends React.Component<ProgramProps, ProgramState> {
   }
 
   _runEffect(effect: Effect): void {
-    if (this.props.debugEnabled && effect !== Effect.empty()) {
-      console.log("[Gongfu] Running Effect");
-    }
-    effect.run(this._updater);
+    if (isEmpty(effect)) return;
+    if (this.props.debugEnabled) console.log("[Gongfu] Running Effect");
+
+    // Run the effect in the next frame (60fps) so React can update the UI in the current event loop
+    setTimeout(effect.run.bind(effect, this._updater), 16);
```

Comment:
Would `setImmediate` or `requestAnimationFrame` be more appropriate than `setTimeout`?

Reply:
`setImmediate` runs at the end of the current event loop, so it wouldn't free the JS thread.
`requestAnimationFrame` seemed like the wrong function because it's related to managing animations (at least in the browser). In React Native, it's implemented with a `setTimeout(fn, 1)`. I can change the timeout length if you prefer?